data/src/main/java/edu/nextstep/camp/calculator/data/CalculationRecord.kt

```diff
@@ -0,0 +1,26 @@
+package edu.nextstep.camp.calculator.data
+
+import io.realm.RealmList
+import io.realm.RealmObject
+
+/**
+ * 클래스에 대한 간단한 설명이나 참고 url을 남겨주세요.
+ * Created by jeongjinhong on 2022. 07. 31..
+ */
+open class CalculationRecord(
+    val calculationRecordList: RealmList<CalculationRecordItem> = RealmList()
+) : RealmObject() {
+    fun addCalculationRecord(expression: String, result: Int) {
+        calculationRecordList.add(CalculationRecordItem(expression, result))
+    }
+}
+
+open class CalculationRecordItem() : RealmObject() {
+    var expression: String? = null
+    var result: Int? = null
+
+    constructor(expression: String, result: Int) : this() {
+        this.expression = expression
+        this.result = result
+    }
+}
```

Comment:
한개의 클래스를 한개의 파일로 나눠주세요.
한개의 파일은 한개의 클래스만 갖는게 좋습니다. 😘

Reply:
클래스 설계시 네이밍에 대해 한번 고민해보셨으면 하는 의견을 드립니다.
클린코드 네이밍 편에 보면 Data, Info와 같은 애매한 단어(저의 표현은 있으나 마나한...)는 지양하라고 정의하고 있습니다.

CalculationRecord와 CalculationRecordItem를 복수와 한개로 의미를 부여하셨는데요.
영어적 표현으로 본다면 둘은 서로 달라보이지 않습니다. Item이 붙어있는것도 역시 의미적으로 달라지거나 하지 않구요.

사용하신 함수에서도 addCalculationRecord 안에서 CalculationRecordItem을 만들어 추가하셨는데요.
무의식적으로 CalculationRecord와 CalculationRecordItem을 동일시 하고 계신게 아닐까 생각이 듭니다. 🤔
 
저는 개인적으로 CalculationRecord와 CalculationRecords(일급컬렉션) 형태로 자주 사용합니다.
일급컬렉션 형태가 아닌 그냥 List를 표현할때는 접미사로 List를 붙입니다. 😊

```suggestion
open class CalculationRecords(
    val calculationRecordList: RealmList<CalculationRecord> = RealmList()
) : RealmObject() {
    fun addCalculationRecord(expression: String, result: Int) {
        calculationRecordList.add(CalculationRecord(expression, result))
    }
}

open class CalculationRecord() : RealmObject() {
    var expression: String? = null
    var result: Int? = null

    constructor(expression: String, result: Int) : this() {
        this.expression = expression
        this.result = result
    }
} 
```